src/main/java/net/minecraftforge/common/extensions/IForgeJukeboxBlock.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.animal.Parrot;
+import net.minecraft.world.entity.animal.allay.Allay;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.RecordItem;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.JukeboxBlock;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.JukeboxBlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Extension-Interface providing methods for various jukebox activiites that are currently hardcoded to only the vanilla jukebox.
+ * Any @{@link Block} implementing will be treated as if it was a jukebox.
+ *
+ * @see JukeboxBlock
+ */
+public interface IForgeJukeboxBlock
+{
+
+    /**
+     * Called by @{@link RecordItem} to insert a record into the jukebox. The actual action taken by the jukebox is up to it and not the record.
+     *
+     * @param entity The entity inserting into jukebox. This can be null.
+     * @param level The level of the jukebox.
+     * @param pos The position of the jukebox.
+     * @param state The blockstate of the jukebox.
+     * @param stack The itemstack being inserted.
+     */
+    default void insertRecord(@Nullable Entity entity, LevelAccessor level, BlockPos pos, BlockState state, ItemStack stack)
```

Comment:
Just for consistency with other block methods, I would reorder this to `BlockState`, `LevelAccessor`, `BlockPos`, `Entity`, `ItemStack`.

Also, what's the argument for making it a `LevelAccessor` compared to a `Level`?